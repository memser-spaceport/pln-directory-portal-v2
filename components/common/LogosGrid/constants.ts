export const LOGOS = [
  '/icons/demoday/landing/logos/Protocol Labs.svg',
  '/icons/demoday/landing/logos/A16zcrypto.svg',
  '/icons/demoday/landing/logos/Coinbase.svg',
  '/icons/demoday/landing/logos/HashKey.svg',
  '/icons/demoday/landing/logos/BlueYard.svg',

  '/icons/demoday/landing/logos/Multicoin.svg',
  '/icons/demoday/landing/logos/PolychainCapital.svg',
  '/icons/demoday/landing/logos/Y.svg',
  '/icons/demoday/landing/logos/PanteraCapital.svg',
  '/icons/demoday/landing/logos/GreyCroft.svg',

  '/icons/demoday/landing/logos/KleinerPerkins.svg',
  '/icons/demoday/landing/logos/Redpoint.svg',
  '/icons/demoday/landing/logos/UnionVentures.svg',
  '/icons/demoday/landing/logos/Venrock.svg',
  '/icons/demoday/landing/logos/Galaxy.svg',

  '/icons/demoday/landing/logos/InsightPartners.svg',
  '/icons/demoday/landing/logos/JuanBenet.svg',
  '/icons/demoday/landing/logos/BalajiSrinivasan.svg',
  '/icons/demoday/landing/logos/VitalikButerin.svg',
  '/icons/demoday/landing/logos/AnatolyYakovenko.svg',
  '/icons/demoday/landing/logos/DigitalCurrency.svg',

  '/icons/demoday/landing/logos/FenbushiCapital.svg',
  '/icons/demoday/landing/logos/Hashed.svg',
  '/icons/demoday/landing/logos/TribeCapital.svg',
  '/icons/demoday/landing/logos/Jump.svg',
  '/icons/demoday/landing/logos/Sosv.svg',

  '/icons/demoday/landing/logos/CoinFund.svg',
  '/icons/demoday/landing/logos/Amber.svg',
  '/icons/demoday/landing/logos/AngelList.svg',
  '/icons/demoday/landing/logos/AnimocaBrands.svg',
  '/icons/demoday/landing/logos/DelphiVentures.svg',

  '/icons/demoday/landing/logos/GGVCapital.svg',
  '/icons/demoday/landing/logos/IOSGVentures.svg',
  '/icons/demoday/landing/logos/Republic.svg',
  '/icons/demoday/landing/logos/SolanaVentures.svg',
  '/icons/demoday/landing/logos/Spartan.svg',

  '/icons/demoday/landing/logos/SVAngel.svg',
  '/icons/demoday/landing/logos/ElectricCapital.svg',
  '/icons/demoday/landing/logos/OrangeDAO.svg',
  '/icons/demoday/landing/logos/Alliance.svg',
  '/icons/demoday/landing/logos/BigBrainHolding.svg',

  '/icons/demoday/landing/logos/Collabcurrency.svg',
  '/icons/demoday/landing/logos/Mesh.svg',
  '/icons/demoday/landing/logos/Placeholder.svg',
  '/icons/demoday/landing/logos/ShimaCapital.svg',
  '/icons/demoday/landing/logos/NorthZone.svg',

  '/icons/demoday/landing/logos/FJLabs.svg',
  '/icons/demoday/landing/logos/Dialectic.svg',
  '/icons/demoday/landing/logos/Arca.svg',
  '/icons/demoday/landing/logos/Borderless.svg',
  '/icons/demoday/landing/logos/MechanismCapital.svg',

  '/icons/demoday/landing/logos/HiveMind.svg',
  '/icons/demoday/landing/logos/FabricVentures.svg',
  '/icons/demoday/landing/logos/Framework.svg',
  '/icons/demoday/landing/logos/JustinKan.svg',
  '/icons/demoday/landing/logos/ChrisLarsen.svg',

  '/icons/demoday/landing/logos/GSR.svg',
  '/icons/demoday/landing/logos/Techstars.svg',
  '/icons/demoday/landing/logos/LongHashVentures.svg',
  '/icons/demoday/landing/logos/Eniac.svg',
  '/icons/demoday/landing/logos/LDCapital.svg',

  '/icons/demoday/landing/logos/NIV.svg',
  '/icons/demoday/landing/logos/LightspeedAction.svg',
  '/icons/demoday/landing/logos/CmtDigital.svg',
  '/icons/demoday/landing/logos/Archetype.svg',
  '/icons/demoday/landing/logos/KrakenVentures.svg',

  '/icons/demoday/landing/logos/OutlinerVentures.svg',
];
