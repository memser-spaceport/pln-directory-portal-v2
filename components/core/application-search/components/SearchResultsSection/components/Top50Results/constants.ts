export const MAX_ITEMS_TO_SHOW = 5;
