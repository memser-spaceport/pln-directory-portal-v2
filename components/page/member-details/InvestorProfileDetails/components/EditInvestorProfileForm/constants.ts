export const investorTypeOptions = [
  { label: 'I angel invest', value: 'ANGEL' },
  { label: 'I invest through fund(s)', value: 'FUND' },
  { label: 'I angel invest + invest through fund(s)', value: 'ANGEL_AND_FUND' },
];

export const investInVcFundsOptions = [
  { label: 'Early stage', value: 'early-stage' },
  { label: 'Late stage', value: 'late-stage' },
  { label: 'Fund-of-funds', value: 'fund-of-funds' },
];

export const SEC_RULES_URL = 'https://www.investor.gov/introduction-investing/general-resources/news-alerts/alerts-bulletins/investor-bulletins/updated-3';
export const EMAIL_SETTINGS_URL = '/settings/email';
export const TEAMS_ADD_URL = '/teams/add';
