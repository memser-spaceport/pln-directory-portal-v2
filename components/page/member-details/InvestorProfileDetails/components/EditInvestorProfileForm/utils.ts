/**
 * Formats a number to currency format
 */
export const formatNumberToCurrency = (value?: string | number): string => {
  if (!value) return '';

  // Convert to number if it's a string
  const numericValue = typeof value === 'string' ? parseFloat(value) : value;
  
  // Return empty string if not a valid number
  if (isNaN(numericValue)) return '';

  // Format as currency
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  }).format(numericValue);
};
