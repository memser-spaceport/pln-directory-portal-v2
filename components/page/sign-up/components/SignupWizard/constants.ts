export const TERMS_OF_SERVICE_AND_PRIVACY_URL =
  'https://docs.google.com/document/d/1J1rYRvnkqzXOnbBoQ3PzjIrSZOTufAL4CJPoOBDq-fE/edit';
