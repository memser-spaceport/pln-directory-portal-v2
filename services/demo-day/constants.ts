export enum DemoDayQueryKeys {
  GET_DEMO_DAY_STATE = 'GET_DEMO_DAY_STATE',
  GET_FUNDRAISING_PROFILE = 'GET_FUNDRAISING_PROFILE',
  GET_TEAMS_LIST = 'GET_TEAMS_LIST',
  GET_ALL_FUNDRAISING_PROFILES = 'GET_ALL_FUNDRAISING_PROFILES',
  GET_ENGAGEMENT = 'GET_ENGAGEMENT',
}
