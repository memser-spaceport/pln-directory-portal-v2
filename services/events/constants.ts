export enum EventsQueryKeys {
  GET_UPCOMING_EVENTS = 'GET_UPCOMING_EVENTS',
}
