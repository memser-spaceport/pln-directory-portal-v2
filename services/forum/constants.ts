export enum ForumQueryKeys {
  GET_CATEGORIES = 'GET_CATEGORIES',
  GET_TOPICS = 'GET_TOPICS',
  GET_TOPIC = 'GET_TOPIC',
  GET_VOTE_STATUS = 'GET_VOTE_STATUS',
}
