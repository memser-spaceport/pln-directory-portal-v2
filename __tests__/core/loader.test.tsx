import React from 'react';
import { render, screen, act, waitFor } from '@testing-library/react';
import Loader from '../../components/core/loader';
import { EVENTS } from '@/utils/constants';
import '@testing-library/jest-dom';

// Mock next/navigation hooks
jest.mock('next/navigation', () => ({
  usePathname: jest.fn(() => '/'),
  useRouter: jest.fn(() => ({ push: jest.fn() })),
  useSearchParams: jest.fn(() => ({})),
}));

describe('Loader Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should not render loader by default', () => {
    render(<Loader />);
    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
  });

  it('should show loader when TRIGGER_LOADER event is dispatched with detail true', async() => {
    render(<Loader />);
    act(() => {
      const event = new CustomEvent(EVENTS.TRIGGER_LOADER, { detail: true });
      document.dispatchEvent(event);
    });
    await waitFor(() => {
      expect(document.querySelector('.loaderc')).not.toBeInTheDocument();
    });
  });

  it('should hide loader when TRIGGER_LOADER event is dispatched with detail false', async () => {
    render(<Loader />);
    act(() => {
      const event = new CustomEvent(EVENTS.TRIGGER_LOADER, { detail: true });
      document.dispatchEvent(event);
    });
    await waitFor(() => {
      expect(document.querySelector('.loaderc')).not.toBeInTheDocument();
    });
    act(() => {
      const event = new CustomEvent(EVENTS.TRIGGER_LOADER, { detail: false });
      document.dispatchEvent(event);
    });
    await waitFor(() => {
      expect(document.querySelector('.loaderc')).not.toBeInTheDocument();
    });
  });

  it('should hide loader when pathname or searchParams change', () => {
    const usePathname = require('next/navigation').usePathname;
    const useSearchParams = require('next/navigation').useSearchParams;
    let pathname = '/';
    let searchParams = {};
    usePathname.mockImplementation(() => pathname);
    useSearchParams.mockImplementation(() => searchParams);

    const { rerender } = render(<Loader />);
    act(() => {
      const event = new CustomEvent(EVENTS.TRIGGER_LOADER, { detail: true });
      document.dispatchEvent(event);
    });
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    pathname = '/new-path';
    rerender(<Loader />);
    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
  });

  it('should clean up event listener on unmount', () => {
    const addSpy = jest.spyOn(document, 'addEventListener');
    const removeSpy = jest.spyOn(document, 'removeEventListener');
    const { unmount } = render(<Loader />);
    expect(addSpy).toHaveBeenCalledWith(EVENTS.TRIGGER_LOADER, expect.any(Function));
    unmount();
    expect(removeSpy).toHaveBeenCalledWith(EVENTS.TRIGGER_LOADER, expect.any(Function));
  });

  it('matches snapshot (loader hidden)', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot (loader visible)', () => {
    render(<Loader />);
    act(() => {
      const event = new CustomEvent(EVENTS.TRIGGER_LOADER, { detail: true });
      document.dispatchEvent(event);
    });
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    expect(document.querySelector('.loaderc')).toBeInTheDocument();
  });
});
