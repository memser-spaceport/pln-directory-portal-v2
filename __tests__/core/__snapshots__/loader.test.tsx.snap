// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component matches snapshot (loader hidden) 1`] = `<DocumentFragment />`;
